fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java

```diff
@@ -178,7 +178,58 @@ private LoanScheduleModel generate(final MathContext mc, final LoanApplicationTe
                 isFirstRepayment = false;
             }
         }
-        while (!scheduleParams.getOutstandingBalance().isZero() || !scheduleParams.getDisburseDetailMap().isEmpty()) {
+
+        Money totalDownPaymentAmount = Money.zero(loanApplicationTerms.getCurrency());
+
+        if (loanApplicationTerms.isDownPaymentEnabled()) {
+            if (!scheduleParams.getDisburseDetailMap().isEmpty()) {
+                for (Map.Entry<LocalDate, Money> disburseDetail : scheduleParams.getDisburseDetailMap().entrySet()) {
+                    BigDecimal downPaymentAmount = disburseDetail.getValue().getAmount()
```

Comment:
MathUtil has `percentageOf` method for this.

Reply:
fixed